src/entities/mellow/mellowLpRouter.ts

```diff
@@ -215,6 +219,22 @@ class MellowLpRouter {
     return getTokenInfo(this.readOnlyContracts.token.address).decimals;
   }
 
+  public get depositable(): boolean {
+    const latestMaturity = this.metadata.vaults.reduce(
+      (latest, vault) => Math.max(latest, vault.maturityTimestampMS),
+      0,
+    );
+    return !this.metadata.deprecated && Date.now().valueOf() + DEPOSIT_WINDOW < latestMaturity;
+  }
+
+  public get withdrawable(): boolean {
```

Comment:
this should get the index of the vault to be withdrawn from, right?

Reply:
resolved.